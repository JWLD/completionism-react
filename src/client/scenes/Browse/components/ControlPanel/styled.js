import styled from 'styled-components'
import FaAlignJustify from 'react-icons/lib/fa/align-justify'
import FaTh from 'react-icons/lib/fa/th'

import { mixins, vars } from 'style'

export const SettingRow = styled.div`
  ${mixins.flexLeft};

  height: ${vars.itemHeight}rem;
  min-height: ${vars.itemHeight}rem;
  width: 100%;
  border-radius: 0.4rem;
  background-color: ${vars.barGrey};
`

export const SettingTitle = styled.span`
  margin-left: 1.5rem;
  margin-right: auto;
  color: white;
  font: 2.5rem DinMC;
`

const SettingIcon = styled.i`
  height: ${vars.itemHeight}rem;
  width: ${vars.itemHeight}rem;
  padding: 0.75rem;
  padding-top: 0.9rem;
  color: white;

  :hover {
    cursor: pointer;
  }
`

export const ListIcon = styled(SettingIcon.withComponent(FaAlignJustify))`
  color: ${props => (props.active ? 'white' : 'grey')};
`

export const SquaresIcon = styled(SettingIcon.withComponent(FaTh))`
  color: ${props => (props.active ? 'white' : 'grey')};
`
