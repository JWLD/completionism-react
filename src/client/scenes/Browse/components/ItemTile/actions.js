import * as ACTIONS from 'constants/action_types'

export const setActiveItem = itemId => ({
  type: ACTIONS.SET_ACTIVE_ITEM,
  payload: itemId
})
