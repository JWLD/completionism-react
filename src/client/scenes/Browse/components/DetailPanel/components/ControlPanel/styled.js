import styled from 'styled-components'
import FaCheck from 'react-icons/lib/fa/check'

import { mixins, vars } from 'style'

const settingPadding = vars.itemListPadding * 0.75
const checkBoxHeight = 2.2

export const Setting = styled.div`
  ${mixins.flexVertical};

  width: 100%;
  padding: ${settingPadding}rem;
  border-radius: 0.4rem;
  background-color: #444;
  color: white;
`

export const SettingTitle = styled.span`
  margin: -0.3rem 0 ${settingPadding * 0.5}rem 0;
  font: 2.25rem DinMC;
  text-transform: uppercase;
`

export const Option = styled.button`
  ${mixins.flex};

  justify-content: space-between;
  width: 100%;
  background-color: transparent;
  color: white;
  font: 2.2rem DinRC;
  line-height: 0;

  :not(:last-child) {
    margin-bottom: ${settingPadding * 0.75}rem;
  }
`

export const CheckBox = styled(FaCheck)`
  height: ${checkBoxHeight}rem;
  width: ${checkBoxHeight}rem;
  padding: 0.2rem;
  border-radius: 0.2rem;
  background-color: ${props => (props.active ? vars.pos : vars.mainGrey)};
  color: ${props => (props.active ? 'white' : vars.mainGrey)};
`
