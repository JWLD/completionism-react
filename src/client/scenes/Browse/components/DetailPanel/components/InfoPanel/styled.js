import styled from 'styled-components'
import FaRocket from 'react-icons/lib/fa/rocket'

import { mixins, vars } from 'style'

export const InfoPanel = styled.div`
  ${mixins.flexVerticalTop};

  width: 100%;
  flex-grow: 1;
  overflow: scroll;

  > *:not(:last-child) {
    margin-bottom: ${vars.itemListPadding}rem;
  }
`

export const ItemName = styled.span`
  ${mixins.flex};

  width: 100%;
  color: white;
  font: 2.5rem DinRC;
  white-space: nowrap;
`

export const ItemImage = styled.div.attrs({
  style: ({ imageUrl }) => ({
    backgroundImage: `url(${imageUrl})`
  })
})`
  width: 100%;
  padding-bottom: 100%;
  border-radius: 0.4rem;
  background-color: #444;
  background-size: cover;
  background-position: center;
`

export const WowheadLink = styled.a`
  ${mixins.flexLeft};

  justify-content: space-between;
  width: 100%;
  height: ${vars.itemHeight}rem;
  padding: 1.2rem;
  border-radius: 0.4rem;
  background-color: #444;
  color: white;
  font: 2.5rem DinRC;

  :hover {
    background-color: #555;
  }
`

export const WowheadIcon = styled(FaRocket)`
  height: 100%;
`
