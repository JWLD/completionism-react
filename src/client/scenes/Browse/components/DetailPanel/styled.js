import styled from 'styled-components'
import { FaCog, FaInfoCircle } from 'react-icons/lib/fa'

import { mixins, vars } from 'style'

export const DetailPanel = styled.div`
  ${mixins.flexVerticalTop};

  justify-content: flex-start;
  align-items: flex-start;
  position: sticky;
  top: ${vars.navBarHeight + vars.browsePadding}rem;
  right: ${vars.browsePadding}rem;
  width: 50rem;
  height: calc(100vh - ${vars.navBarHeight + vars.browsePadding * 2}rem);
  padding: ${vars.itemListPadding}rem;
  border-radius: 0.4rem;
  background-color: ${vars.mainGrey};

  > * {
    :not(:last-child) {
      margin-bottom: ${vars.itemListPadding}rem;
    }
  }
`

export const TopBar = styled.div`
  ${mixins.flex};

  height: ${vars.itemHeight}rem;
  min-height: ${vars.itemHeight}rem;
  width: 100%;
`

export const Title = styled.span`
  ${mixins.flex};

  flex-grow: 1;
  height: 100%;
  border-radius: 0.4rem;
  background-color: #444;
  color: white;
  font: 3rem DinMC;
  text-transform: uppercase;
`

export const ToggleWrap = styled.button`
  ${mixins.flex};

  position: relative;
  height: ${vars.itemHeight}rem;
  width: ${vars.itemHeight * 2 - vars.itemHeight * 0.15}rem;
  padding: ${vars.itemHeight * 0.15}rem;
  margin-left: ${vars.itemListPadding}rem;
  border-radius: 0.4rem;
  background-color: ${vars.mainDark};
  overflow: hidden;
`

export const CogIcon = styled(FaCog)`
  position: absolute;
  right: 0;
  top: 0;
  height: ${vars.itemHeight}rem;
  width: ${vars.itemHeight}rem;
  padding: 0.75rem;
  color: ${props => (props.active ? 'white' : 'gray')};
`

export const InfoIcon = CogIcon.withComponent(FaInfoCircle).extend`
  right: auto;
  left: 0;
`
