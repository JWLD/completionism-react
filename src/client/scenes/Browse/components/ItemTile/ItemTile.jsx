import React from 'react'
import PropTypes from 'prop-types'

import { ICON_URLS } from 'constants/urls'
import * as SC from './styled'

const isCollected = props => {
  const { category, id, storageData } = props

  let collected = false

  if (category === 'pets') {
    collected = storageData.find(el => el.id === id)
  } else {
    collected = storageData.includes(id)
  }

  return collected
}

const ItemTile = props => {
  const { icon, name, quality } = props

  const itemIcon = icon
    ? { backgroundImage: `url(${ICON_URLS.large}${icon}.jpg)` }
    : null

  return (
    <SC.ItemTile>
      <SC.ItemIcon style={itemIcon} />
      <SC.ItemTitle quality={quality}>{name}</SC.ItemTitle>
      {isCollected(props) ? <SC.CheckIcon /> : <SC.CrossIcon />}
    </SC.ItemTile>
  )
}

ItemTile.defaultProps = {
  icon: null
}

ItemTile.propTypes = {
  icon: PropTypes.string,
  name: PropTypes.string.isRequired,
  quality: PropTypes.number.isRequired
}

export default ItemTile
