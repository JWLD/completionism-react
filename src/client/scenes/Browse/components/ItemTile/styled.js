import styled from 'styled-components'
import FaCheckCircle from 'react-icons/lib/fa/check-circle'
import FaTimesCircle from 'react-icons/lib/fa/times-circle'

import { mixins, vars } from 'style'

const itemTilePadding = 0.5
const iconSize = vars.sizing.itemHeight - itemTilePadding * 2

export const ItemTile = styled.li`
  ${mixins.itemListItem};

  justify-content: space-between;
  height: ${vars.sizing.itemHeight}rem;
  padding: ${itemTilePadding}rem;
  background-color: #444;

  :not(:last-child) {
    margin-bottom: 1rem;
  }

  :hover {
    cursor: pointer;
    background-color: #555;
  }
`

export const ItemIcon = styled.i`
  height: ${iconSize}rem;
  width: ${iconSize}rem;
  margin-right: ${itemTilePadding * 2}rem;

  border-radius: 0.4rem;
  background-size: cover;
  background-color: #222;
`

export const ItemTitle = styled.span`
  flex-grow: 1;
  color: ${props => vars.colours[`q${props.quality}`]};
  font: 2.25rem DinRC;
  text-align: center;
`

export const ProgressIconWrap = styled.div`
  height: ${iconSize}rem;
  width: ${iconSize}rem;
  padding: 0.5rem;
  border-radius: 0.4rem;
  background-color: #333;
`

export const CheckIcon = ProgressIconWrap.withComponent(FaCheckCircle).extend`
  color: ${vars.colours.pos};
`

export const CrossIcon = ProgressIconWrap.withComponent(FaTimesCircle).extend`
  color: ${vars.colours.neg};
`
