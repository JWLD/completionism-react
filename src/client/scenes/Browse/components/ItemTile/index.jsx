import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { ICON_URLS } from 'constants/urls'
import { setActiveItem } from 'ItemTile/actions'

import * as SC from 'ItemTile/styled'

export const ItemTile = props => {
  const { collected, icon, id, name, quality } = props

  const iconUrl = icon ? `url(${ICON_URLS.large}${icon}.jpg)` : 'none'
  const progressIcon = collected ? <SC.CheckIcon /> : <SC.CrossIcon />

  return (
    <SC.ItemTile onClick={() => props.setActiveItem(id)}>
      <SC.ItemIcon iconUrl={iconUrl} />
      <SC.ItemTitle quality={quality}>{name}</SC.ItemTitle>
      {progressIcon}
    </SC.ItemTile>
  )
}

ItemTile.defaultProps = {
  icon: null
}

ItemTile.propTypes = {
  collected: PropTypes.bool.isRequired,
  icon: PropTypes.string,
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  quality: PropTypes.number.isRequired,
  setActiveItem: PropTypes.func.isRequired
}

const mapDispatchToProps = {
  setActiveItem
}

export default connect(null, mapDispatchToProps)(ItemTile)
