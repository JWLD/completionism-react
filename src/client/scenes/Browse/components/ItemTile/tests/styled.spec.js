import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { ItemIcon, ItemTitle } from 'scenes/Browse/components/ItemTile/styled'

Enzyme.configure({ adapter: new Adapter() })

describe('<ItemIcon />', () => {
  const mockProps = {
    iconUrl: 'foo'
  }

  it('handles [iconUrl] prop correctly', () => {
    const component = shallow(<ItemIcon {...mockProps} />)

    expect(component).toMatchSnapshot()
  })
})

describe('<ItemTitle />', () => {
  const mockProps = {
    quality: 1
  }

  it('handles [quality] prop correctly', () => {
    const component = shallow(<ItemTitle {...mockProps} />)

    expect(component).toMatchSnapshot()
  })
})
