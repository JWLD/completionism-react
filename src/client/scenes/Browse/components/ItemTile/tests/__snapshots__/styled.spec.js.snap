// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemIcon /> handles [iconUrl] prop correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.i
    iconUrl="foo"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-bwzfXH ajfbh",
      "style": Object {
        "backgroundImage": "foo",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-bwzfXH ajfbh",
        "style": Object {
          "backgroundImage": "foo",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ItemTitle /> handles [quality] prop correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemTitle
    quality={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-htpNat dphMKM",
    },
    "ref": null,
    "rendered": null,
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-htpNat dphMKM",
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
